### PR TITLE
Do not run "symflower fix" if the original response failed with a timeout, so the model and the fix assessments are consistent

### PR DESCRIPTION
**Pull request by ruiAzevedo19**
Part of #232

### CODE REVIEW

**Comment by ruiAzevedo19** [comment]
@bauersimon I wanted to add a unit test for this case but I think it's not that easy:
- the execution timeout value is hard-coded, so in order to test the changes I would need to change `language.DefaultExecutionTimeout = <timeout-for-tests>` and then reset it to the original value. This approach sounds wrong
- I would need a scenario where the 1st call to the execution function (for the model)  times out, but the 2nd call (for `symflower fix`) does not. Not possible with the current implementation
- Also thought of a system test where we can set the execution time via CLI, but the value is in minutes.

File: evaluate/task/task-write-test.go [hunk 1]
```diff
@@ -97,6 +99,14 @@ func (t *TaskWriteTests) Run(repository evaltask.Repository) (repositoryAssessme
 		if err != nil {
 			problems = append(problems, pkgerrors.WithMessage(err, filePath))
 
+			// If there is an execution timeout do not run "symflower fix".
```
Comment (bauersimon) [inline evaluate/task/task-write-test.go:102]: `cause the code itself is correct`.

**Comment by bauersimon** [comment]
I think one clean possibility would be to have the execution timeout as a method of the language interface and then use a mock of a language that can return a lower value. Can you try that out and if it takes longer then 10 minutes we just go with what we have? It's just an optimization for like a 1% chance. So not super duper important to test if it's such a hassle.

**Comment by ruiAzevedo19** [comment]
> I think one clean possibility would be to have the execution timeout as a method of the language interface and then use a mock of a language that can return a lower value. Can you try that out and if it takes longer then 10 minutes we just go with what we have? It's just an optimization for like a 1% chance. So not super duper important to test if it's such a hassle.

Did it a bit differently: I mocked the `Execute` function to return the timeout error and did the assertions accordingly. Reason being the timeout method would be called inside the `Execute` method, which needs also to be mocked.

File: evaluate/task/task-write-test_test.go [hunk 2]
```diff
@@ -178,12 +182,42 @@ func TestTaskWriteTestsRun(t *testing.T) {
...
+				}
+				expectedProblems := []string{
+					"context deadline exceeded",
+				}
+
+				languageMock := languagetesting.NewMockLanguageNamed(t, "golang")
+				languageMock.On("Files", mock.Anything, mock.Anything).Return([]string{filepath.Join("golang", "plain")}, nil).Once()
+				languageMock.On("Execute", mock.Anything, mock.Anything).Return(uint64(0), nil, context.DeadlineExceeded).Once()
+
+				validateGo(t, "Model generated test with unused import", languageMock, bytesutil.StringTrimIndentations(`
```
Comment (bauersimon) [inline evaluate/task/task-write-test_test.go:208]: update test name to reflect timeout

File: evaluate/task/task-write-test_test.go [hunk 3]
```diff
@@ -178,12 +182,42 @@ func TestTaskWriteTestsRun(t *testing.T) {
...
+					package plain
+
+					import (
+						"testing"
+						"strings"
+					)
+
+					func TestPlain(t *testing.T) {
+					   	plain()
+					}
```
Comment (bauersimon) [inline evaluate/task/task-write-test_test.go:218]: Please make this an empty string so it's clear that it is not used.

**Review by bauersimon — CHANGES_REQUESTED** [review]

**Review by bauersimon — APPROVED** [review]

**Pull request merged**